mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -0,0 +1,251 @@
+        :return: A single object of :py:class:`mlflow.entities.model_registry.RegisteredModel`
+                 created in the backend.
+        """
+        req_body = message_to_json(CreateRegisteredModelRequest(name=name, description=description))
+        response_proto = self._call_endpoint(CreateRegisteredModelRequest, req_body)
+        return registered_model_from_uc_proto(response_proto.registered_model)
+
+    def update_registered_model(self, name, description):
+        """
+        Update description of the registered model.
+
+        :param name: Registered model name.
+        :param description: New description.
+        :return: A single updated :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        req_body = message_to_json(UpdateRegisteredModelRequest(name=name, description=description))
+        response_proto = self._call_endpoint(UpdateRegisteredModelRequest, req_body)
+        return registered_model_from_uc_proto(response_proto.registered_model)
+
+    def rename_registered_model(self, name, new_name):
+        """
+        Rename the registered model.
+
+        :param name: Registered model name.
+        :param new_name: New proposed name.
+        :return: A single updated :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        _raise_unsupported_method(
+            method="rename_registered_model",
+            message="Use the Unity Catalog REST API to rename registered models",
+        )
+
+    def delete_registered_model(self, name):
+        """
+        Delete the registered model.
+        Backend raises exception if a registered model with given name does not exist.
+
+        :param name: Registered model name.
+        :return: None
+        """
+        req_body = message_to_json(DeleteRegisteredModelRequest(name=name))
+        self._call_endpoint(DeleteRegisteredModelRequest, req_body)
+
+    def search_registered_models(
+        self, filter_string=None, max_results=None, order_by=None, page_token=None
+    ):
+        """
+        Search for registered models in backend that satisfy the filter criteria.
+
+        :param filter_string: Filter query string, defaults to searching all registered models.
+        :param max_results: Maximum number of registered models desired.
+        :param order_by: List of column names with ASC|DESC annotation, to be used for ordering
+                         matching search results.
+        :param page_token: Token specifying the next page of results. It should be obtained from
+                            a ``search_registered_models`` call.
+        :return: A PagedList of :py:class:`mlflow.entities.model_registry.RegisteredModel` objects
+                that satisfy the search expressions. The pagination token for the next page can be
+                obtained via the ``token`` attribute of the object.
+        """
+        _require_arg_unspecified("filter_string", filter_string)
```

Comment:
Filtering down the list of registered models & specifying a custom ordering is unsupported initially